Std/Data/Function/Basic.lean

```diff
@@ -0,0 +1,124 @@
+/-
+Copyright (c) Mathlib contributors.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl, Mario Carneiro
+-/
+import Std.Logic
+
+namespace Function
+
+theorem funext_iff {β : α → Sort _} {f₁ f₂ : ∀ x : α, β x} : f₁ = f₂ ↔ ∀ a, f₁ a = f₂ a :=
+  Iff.intro (fun h _ ↦ h ▸ rfl) funext
+
+section Update
+
+variable {α : Sort u} {β : α → Sort v} {α' : Sort w} [DecidableEq α] [DecidableEq α']
+  {f g : (a : α) → β a} {a : α} {b : β a}
+
+/-- Replacing the value of a function at a given point by a given value. -/
+def update (f : ∀ a, β a) (a' : α) (v : β a') (a : α) : β a :=
+  if h : a = a' then Eq.ndrec v h.symm else f a
+
+/-- On non-dependent functions, `Function.update` can be expressed as an `ite` -/
+theorem update_apply {β : Sort _} (f : α → β) (a' : α) (b : β) (a : α) :
+    update f a' b a = if a = a' then b else f a :=
+by have h2 : (h : a = a') → Eq.rec (motive := λ _ _ => β) b h.symm = b :=
+     by intro h; cases h; simp
+   have h3 : (λ h : a = a' => Eq.rec (motive := λ _ _ => β) b h.symm) =
+             (λ _ : a = a' =>  b) := funext h2
+   let f := λ x => dite (a = a') x (λ (_: ¬ a = a') => (f a))
+   exact congrArg f h3
+
+@[simp]
+theorem update_same (a : α) (v : β a) (f : ∀ a, β a) : update f a v a = v :=
+  dif_pos rfl
+
+@[simp]
+theorem update_noteq {a a' : α} (h : a ≠ a') (v : β a') (f : ∀ a, β a) : update f a' v a = f a :=
```

Comment:
`update_ne` would be just as clear. If there is a conflict I'm not seeing, consider `update_other`.

Reply:
These were the names in Mathlib. I agree that `update_ne` and `update_eq` are more consistent with other functions with similar statements, so we should change it IMO.